setup.py

```diff
@@ -27,28 +23,59 @@ def read(filename):
     'pytest-html',
     'pytest-cov',
     'pytest-helpers-namespace',
-    'pytest >= 3.0'
+    'pytest >= 3.0',
+    'gitpython'
 ]
 
-setup(
-    name="k8s",
-    author="FINN Team Infrastructure",
-    author_email="FINN-TechteamInfrastruktur@finn.no",
-    version="1.0",
-    packages=find_packages(exclude=("tests",)),
-    zip_safe=True,
-    include_package_data=True,
-
-    # Requirements
-    install_requires=GENERIC_REQ,
-    setup_requires=['pytest-runner', 'wheel', 'setuptools_git >= 0.3'],
-    extras_require={
-        "dev": TESTS_REQ + CODE_QUALITY_REQ,
-        "codacy": ["codacy-coverage"]
-    },
-
-    # Metadata
-    description="Python client library for the Kubernetes API",
-    long_description=read("README.md"),
-    url="https://github.com/fiaas/k8s",
-)
+
+def _has_tags():
+    """setuptools_git will crash if you ask it to use vcs version without any tags in the repo"""
+    try:
+        subprocess.check_output(["git", "describe"], stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        return False
+    return True
+
+
+def _generate_description():
+    description = [_read("README.md")]
+    changelog_file = os.getenv("CHANGELOG_FILE")
+    if changelog_file:
+        description.append(_read(changelog_file))
+    return "\n".join(description)
+
+
+def _read(filename):
+    with open(os.path.join(os.path.dirname(__file__), filename)) as f:
+        return f.read()
+
+
+def main():
+    setup(
+        name="k8s",
+        author="FINN Team Infrastructure",
+        author_email="FINN-TechteamInfrastruktur@finn.no",
```

Comment:
Should we maybe remove this untill we have a better contact point?

Reply:
On the other hand, at least here we have a contact point that reaches the devs, even if it might be a bit off target? I don't think it's really important at this point.

Anyway, I have a plan to go over the metadata in a different PR before creating the first release. We need some classifiers and a few other metadata items I think.